Weaver/WeaverTests/RealmWeaver.Tests/WeaveEmptyTests.cs

```diff
@@ -0,0 +1,76 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Reflection;
+using Mono.Cecil;
+using NUnit.Framework;
+
+namespace RealmWeaver 
+{
+    [TestFixture(AssemblyType.NonPCL)]
+    [TestFixture(AssemblyType.PCL)]
+    public class WeaveEmptyTests : WeaverTestBase 
+    {
+        private string _sourceAssemblyPath;
+
+        public WeaveEmptyTests(AssemblyType assemblyType) : base(assemblyType)
+        {
+        }
+
+        [OneTimeSetUp]
+        public void FixtureSetup()
+        {            
+            _sourceAssemblyPath = _assemblyType == AssemblyType.NonPCL
```

Comment:
The majority of this method can be reused if you extract it to the base class. You only need to expose two abstract properties for the locator types.